src/blueapi/worker/reworker.py

```diff
@@ -95,17 +96,17 @@ def clear_task(self, task_id: str) -> str:
 
     def cancel_active_task(
         self,
-        cancel_type: WorkerState,
+        failure: bool = False,
         reason: Optional[str] = None,
     ) -> str:
         if self._current is None:
-            raise KeyError("Worker has no active task")
-        if self._current.is_complete:
-            raise TypeError("Task already complete")
-        if cancel_type is WorkerState.STOPPING:
-            self._ctx.run_engine.stop()
-        if cancel_type is WorkerState.ABORTING:
+            # Persuades mypy that self._current is not None
+            # We only allow this method to be called if a Plan is active
+            raise TransitionError
```

Comment:
Should this be an instance rather than the class?

Reply:
raise XError is equivalent to raise XError()

I've added a message to the exception just to make it distinct from the TransitionError in the RunEngine that it prevents us reaching, in case we ever call not from the REST client that immediately catches it.